### ISSUE TITLE
Enums generation not working for single null value

### ISSUE THREAD

**Issue by az0uz** [anymaniax/orval#750]
Hi thanks for the great library, I have encountered an issue from a model generated by django db model, using a `django.db.model.CharField(null=True, blank=True, choices=FooTextChoices.choices)`. See the following test schema you could add to the project tests.

Part of this issue is #605 but it also fails to generate enums with a single `null` value

## What are the steps to reproduce this issue?

1. Add the following test configuration in `tests/configs/react-query.config.ts`

```typescript
export default defineConfig({
  ...
  polymorphicEnum: {
    output: {
      target: '../generated/react-query/polymorphic-enum/endpoints.ts',
      schemas: '../generated/react-query/polymorphic-enum/model',
      client: 'react-query',
      mock: true,
      headers: true,
    },
    input: {
      target: '../specifications/polymorphic-enum.yaml',
    },
  },
});
```

2. Add the test schema at `tests/specificationspolymorphic-enum.yaml`

```yaml
openapi: 3.0.3
info:
  title: API
  version: V1
paths:
  /demo:
    get:
      operationId: getPolymorphicEnumResponse
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/BlankableNullableEnum'
components:
  schemas:
    BlankableNullableEnum:
      nullable: true
      oneOf:
        - $ref: '#/components/schemas/NullEnum'
        - $ref: '#/components/schemas/BlankEnum'
        - $ref: '#/components/schemas/SampleEnum'
    NullEnum:
      enum:
      - null
    BlankEnum:
      enum:
      - ''
    SampleEnum:
      enum:
      - A
      - B
      - C
      type: string

```

## What happens?

The generator generated a broken model:

- In `tests/generated/react-query/polymorphic-enum/model/nullEnum.ts`

```typescript
export interface NullEnum 'null' // Parsing error: '{' expected.
```

- In `tests/generated/react-query/polymorphic-enum/model/blankEnum.ts`

```typescript
export interface BlankEnum '' // Parsing error: '{' expected.
```

## What were you expecting to happen?

- In `tests/generated/react-query/polymorphic-enum/model/nullEnum.ts`

```typescript
export type NullEnum = typeof NullEnum[keyof typeof NullEnum];

export const NullEnum = {
  null: 'null',
}
```

- In `tests/generated/react-query/polymorphic-enum/model/blankEnum.ts`

```typescript
export type BlankEnum = typeof BlankEnum[keyof typeof BlankEnum]

export const BlankEnum = {
  '': '',
} as const

```

## Any logs, error output, etc?

- forcing the BlankEnum to type string seems to fix it, which is the bug already reported in #605

```typescript
export type BlankEnum = typeof BlankEnum[keyof typeof BlankEnum];

export const BlankEnum = {
  '': '',
} as const;

```

- NullEnum won't generate if `type='string'` and `nullable=true`: `polymorphicEnum - TypeError: Cannot read properties of null (reading 'replace')`

- NullEnum will generate a broken schema if `type='null'`:

```typescript
export interface NullEnum 'null' // Parsing error: '{' expected.
```

- NullEnum will generate a broken schema if `type='null'` and `nullable=true`:

```typescript
export type NullEnum = typeof NullEnum[keyof typeof NullEnum];

export const NullEnum = {
  null: 'null',
  null: null, // Error: An object literal cannot have multiple properties with the same name.
} as const;
```

## Any other comments?

It might be related to this openapi spec clarification proposal: https://github.com/OAI/OpenAPI-Specification/blob/main/proposals/2019-10-31-Clarify-Nullable.md

The inferred type of NullEnum should maybe be 'null', and the generator work for both nullable as true or false.

## What versions are you using?

**Operating System:** MacOS 13.0.1
**Package Version:** v6.11.1
**Browser Version:** None

**Comment by krzyswidlo** [comment]
Bump, I'm encountering the same issue in node.js. Enum defined as nullable and containing null value is generated with two null properties.

**Comment by melloware** [comment]
Looks like this is resolved try 6.19.1 and report back if it does not work.
